### PR TITLE
Align Terraform  config with current state

### PR DESCRIPTION
**Pull request by aaclan-ebi**
Resolving the following conflicts:

 ```
  ~ module.upload-service.aws_lambda_function.area_deletion_lambda
      tags.%:                            "4" => "0"
      tags.Owner:                        "tburdett" => ""
      tags.Project:                      "hca" => ""
      tags.Service:                      "ait" => ""
      tags.production:                   "dev" => ""
  ~ module.upload-service.aws_lambda_function.upload_api_lambda
      tags.%:                            "5" => "1"
      tags.Owner:                        "tburdett" => ""
      tags.Project:                      "hca" => ""
      tags.Service:                      "ait" => ""
      tags.environment:                  "dev" => ""
  ~ module.upload-service.aws_lambda_function.upload_checksum_lambda
      tags.%:                            "4" => "0"
      tags.Owner:                        "tburdett" => ""
      tags.Project:                      "hca" => ""
      tags.Service:                      "ait" => ""
      tags.environment:                  "dev" => ""
  ~ module.upload-service.aws_lambda_function.validation_scheduler_lambda
      tags.%:                            "4" => "0"
      tags.Owner:                        "tburdett" => ""
      tags.Project:                      "hca" => ""
      tags.Service:                      "ait" => ""
      tags.environment:                  "dev" => ""
  ~ module.upload-service.aws_s3_bucket.lambda_deployments
      tags.%:                            "1" => "0"
      tags.Project:                      "hca" => ""
  ~ module.upload-service.aws_s3_bucket.upload_areas_bucket
      tags.%:                            "1" => "0"
      tags.Project:                      "hca" => ""
-/+ module.upload-service.aws_secretsmanager_secret_version.secrets (new resource required)
      id:                                "dcp/upload/dev/secrets|35AEAEA6-4622-4A9C-A1D6-65948EC74BA6" => <computed> (forces new resource)
      arn:                               "arn:aws:secretsmanager:us-east-1:871979166454:secret:dcp/upload/dev/secrets-JbuXao" => <computed>
      secret_id:                         "dcp/upload/dev/secrets" => "dcp/upload/dev/secrets"
      secret_string:                     <sensitive> => <sensitive> (forces new resource)
      version_id:                        "35AEAEA6-4622-4A9C-A1D6-65948EC74BA6" => <computed>
      version_stages.#:                  "1" => <computed>
  ~ module.upload-service.module.upload-service-database.aws_ecs_cluster.pgbouncer
      tags.%:                            "4" => "0"
      tags.Name:                         "upload-service" => ""
      tags.Owner:                        "tburdett" => ""
      tags.Project:                      "hca" => ""
      tags.Service:                      "ait" => ""
  ~ module.upload-service.module.upload-service-database.aws_lb.main
      tags.%:                            "4" => "0"
      tags.Name:                         "upload-pgbouncer" => ""
      tags.Owner:                        "tburdett" => ""
      tags.Project:                      "hca" => ""
      tags.Service:                      "ait" => ""
  ~ module.upload-service.module.upload-service-database.aws_rds_cluster.upload
      db_cluster_parameter_group_name:   "default.aurora-postgresql10" => "default.aurora-postgresql9.6"
      engine_version:                    "10.11" => "9.6.12"
      tags.%:                            "4" => "0"
      tags.Name:                         "upload" => ""
      tags.Owner:                        "tburdett" => ""
      tags.Project:                      "hca" => ""
      tags.Service:                      "ait" => ""
-/+ module.upload-service.module.upload-service-database.aws_rds_cluster_instance.cluster_instances (new resource required)
      id:                                "upload-cluster-dev-0" => <computed> (forces new resource)
      apply_immediately:                 "" => <computed>
      arn:                               "arn:aws:rds:us-east-1:871979166454:db:upload-cluster-dev-0" => <computed>
      auto_minor_version_upgrade:        "true" => "true"
      availability_zone:                 "us-east-1a" => <computed>
      ca_cert_identifier:                "rds-ca-2019" => <computed>
      cluster_identifier:                "upload-dev" => "upload-dev"
      copy_tags_to_snapshot:             "false" => "false"
      db_parameter_group_name:           "default.aurora-postgresql10" => <computed>
      db_subnet_group_name:              "upload_db_subnet_group_dev" => <computed>
      dbi_resource_id:                   "db-YKQBXHOLAHNVRNC3T5ZW4XLOOQ" => <computed>
      endpoint:                          "upload-cluster-dev-0.cdbjygwfgkap.us-east-1.rds.amazonaws.com" => <computed>
      engine:                            "aurora-postgresql" => "aurora-postgresql"
      engine_version:                    "10.11" => "9.6.12" (forces new resource)
      identifier:                        "upload-cluster-dev-0" => "upload-cluster-dev-0"
      identifier_prefix:                 "" => <computed>
      instance_class:                    "db.t3.medium" => "db.r4.large"
      kms_key_id:                        "arn:aws:kms:us-east-1:871979166454:key/a5d6dbe1-5892-41ff-9ebf-c5cce8def489" => <computed>
      monitoring_interval:               "0" => "0"
      monitoring_role_arn:               "" => <computed>
      performance_insights_enabled:      "true" => "true"
      performance_insights_kms_key_id:   "arn:aws:kms:us-east-1:871979166454:key/a5d6dbe1-5892-41ff-9ebf-c5cce8def489" => <computed>
      port:                              "5432" => <computed>
      preferred_backup_window:           "07:27-07:57" => <computed>
      preferred_maintenance_window:      "sat:09:08-sat:09:38" => "sat:09:08-sat:09:38"
      promotion_tier:                    "0" => "0"
      publicly_accessible:               "true" => "true"
      storage_encrypted:                 "true" => <computed>
      writer:                            "true" => <computed>
```

### CODE REVIEW

**Comment by aaclan-ebi** [comment]
hmm perhaps i need to source the env from a variable ...

File: terraform/modules/upload-service/bucket.tf [hunk 1]
```diff
@@ -3,6 +3,10 @@ resource "aws_s3_bucket" "upload_areas_bucket" {
   acl = "private"
   force_destroy = "false"
   acceleration_status = "Enabled"
+  tags = {
```
Comment (justincc) [inline terraform/modules/upload-service/bucket.tf:6]: Can we also add the Owner, Service and and environment (if applicable) tags like elsewhere?

File: terraform/modules/upload-service/bucket.tf [hunk 2]
```diff
@@ -85,4 +89,7 @@ resource "aws_s3_bucket" "lambda_deployments" {
...
```
Comment (justincc) [inline terraform/modules/upload-service/bucket.tf:94]: Can we also add the Owner, Service and and environment (if applicable) tags like elsewhere?

**Review by justincc — APPROVED** [review]

File: terraform/envs/dev/variables.tf [hunk 3]
```diff
@@ -109,3 +109,18 @@ variable "dcp_auth0_audience" {
 variable "gcp_service_acct_creds" {
   type = "string"
 }
+
+// AWS RDS Cluster instance
+
+variable "aws_rds_cluster_instance_class" {
```
Comment (rolando-ebi) [inline terraform/envs/dev/variables.tf:115]: I think we should git-commit the value we use here. That way devs can review changes to instance sizes, postgres versions, etc. in PRs (the big plus of infrastructure-as-code). But I'm not sure if the terraform deploy scripts for upload-service are set up in that way, from what I remember when working on them

**Review by rolando-ebi — APPROVED** [review]

File: terraform/envs/dev/variables.tf (hunk 3 above)
Comment (aaclan-ebi) [reply terraform/envs/dev/variables.tf:115]: all tfvars are currently being sourced from s3

File: terraform/modules/upload-service/bucket.tf (hunk 2 above)
Comment (aaclan-ebi) [reply terraform/modules/upload-service/bucket.tf:94]: Added in config applied the changes, tags should be added in the resources now.

File: terraform/envs/dev/variables.tf (hunk 3 above)
Comment (aaclan-ebi) [reply terraform/envs/dev/variables.tf:115]: Should be possible to do but maybe that's also intentional. The buckets are versioned and can be reviewed, i guess it's fine for now.

 The changes are:
for dev and staging:
```
// AWS RDS Cluster Instance
aws_rds_cluster_instance_class="db.t3.medium"
aws_rds_cluster_instance_engine_version="10.11"
aws_rds_db_cluster_parameter_group_name="default.aurora-postgresql10"
```
for prod:
```
// AWS RDS Cluster Instance
aws_rds_cluster_instance_class="db.r4.large"
aws_rds_cluster_instance_engine_version="9.6.12"
aws_rds_db_cluster_parameter_group_name="default.aurora-postgresql9.6"
```

File: terraform/modules/upload-service/bucket.tf (hunk 1 above)
Comment (aaclan-ebi) [reply terraform/modules/upload-service/bucket.tf:6]: Added in config applied the changes, tags should be added in the resources now.

**Comment by aaclan-ebi** [comment]
Thanks for reviewing @rolando-ebi and @justincc ! merging now...

**Pull request merged**